compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrInlineCodegen.kt

```diff
@@ -93,7 +93,12 @@ class IrInlineCodegen(
         expression: IrFunctionAccessExpression
     ) {
         val typeArguments = expression.descriptor.typeParameters.keysToMap { expression.getTypeArgumentOrDefault(it) }
-        performInline(typeArguments, callDefault, codegen)
+        state.globalInlineContext.enterIntoInlining(null)
```

Comment:
Please add TODO, that  IrFunctionAccessExpression should be somehow passed for cycle check